backend/prisma/migrations/20220907054238_add_unique_constraints_equipment/migration.sql

```diff
@@ -0,0 +1,12 @@
+/*
+  Warnings:
+
+  - A unique constraint covering the columns `[name]` on the table `Equipment` will be added. If there are existing duplicate values, this will fail.
+  - A unique constraint covering the columns `[serial]` on the table `Equipment` will be added. If there are existing duplicate values, this will fail.
+
+*/
```

Comment:
this would be a problem if we had existing duplicate names/serial numbers in our db. before adding the constraint, we need to somehow delete duplicate values.

can you try adding this before creating the unique indexes:
```sql
DELETE FROM "Equipment" e1
USING "Equipment" e2
WHERE e1.id != e2.id
AND (e1.name= e2.name OR e1.serial = e2.serial);
```
but before you rerun the app, do a `docker compose down -v && docker compose up` and then temporarily delete this migration folder before rerunning the app. then add equipments with duplicate names and serials in the db. after that, you can revive this folder by running `git checkout HEAD backend/prisma/migrations/20220907054238_add_unique_constraints_equipment/migration.sql` if the migration works and deletes the duplicate rows, it should be good to push and merge

Reply:
noted sir